v2/writer.go

```diff
@@ -79,6 +84,109 @@ func WrapV1(src io.ReadSeeker, dst io.Writer) error {
 	return nil
 }
 
+// ExtractV1File takes a CARv2 file and extracts its CARv1 data payload, unmodified.
+// The resulting CARv1 file will not include any data payload padding that may be present in the
+// CARv2 srcPath.
+// If srcPath represents a CARv1 ErrAlreadyV1 error is returned.
+// The srcPath is assumed to exist, and the destination path is created if not exist.
+// Note that the destination path might still be created even if an error
+// occurred.
+// If srcPath and dstPath are the same, then the dstPath is converted, in-place, to CARv1.
+func ExtractV1File(srcPath, dstPath string) (err error) {
+	src, err := os.Open(srcPath)
+	if err != nil {
+		return err
+	}
+
+	// Ignore close error since only reading from src.
+	defer src.Close()
+
+	// Detect CAR version.
+	version, err := ReadVersion(src)
+	if err != nil {
+		return err
+	}
+	if version == 1 {
+		return ErrAlreadyV1
+	}
+	if version != 2 {
+		return fmt.Errorf("invalid source version: %v", version)
```

Comment:
maybe say "source version must be 2" to make it clearer? invalid makes me think a version is truly bad, like -3.